### ISSUE TITLE
[package] ninja/1.9.0: Need to be built by GCC 4.9

### ISSUE THREAD

**Issue by uilianries** [conan-io/conan-center-index#1327]
Ninja package is an installer and is built and distributed using GCC 5 as the compiler setting is removed from its package id. The problem occurs when a package requires Ninja as build requirement, but needs to be build by GCC 4.9. 

The glibc version from Trusty (GCC 4.9) is incompatible to Xenial (GCC 5), which results is linkage error.

We need to find some way to detect when a package is an installer and build using GCC 4.9 to keep the backward compatibility.

/cc @danimtb @solvingj

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **ninja/1.9.0**
  * Operating System+version: **Linux Ubuntu 18.04**
  * Compiler+version: **GCC 5**
  * Docker image: **conanio/gcc5**
  * Conan version: **conan 1.24.0**
  * Python version: **Python 3.8**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=4.9
compiler.libcxx=libstdc++
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)

Build the PR #1325

### Logs (Include/Attach if Applicable)

Check the PR #1325 logs

**Comment by ericLemanissier** [comment]
continuation of https://github.com/conan-io/conan-center-index/issues/624 and https://github.com/conan-io/conan-center-index/issues/213

**Comment by uilianries** [comment]
fixed by #2062
